### PR TITLE
Overload type signature for processMessage

### PR DESCRIPTION
**Pull request by diehuxx**

### CODE REVIEW

File: src/dwn.ts [hunk 1]
```diff
@@ -90,6 +91,19 @@ export class Dwn {
...
+  public async processMessage(tenant: string, rawMessage: ProtocolsConfigureMessage, dataStream?: Readable): Promise<GenericMessageReply>;
+  public async processMessage(tenant: string, rawMessage: ProtocolsQueryMessage, dataStream?: Readable): Promise<ProtocolsQueryReply>;
+  public async processMessage(tenant: string, rawMessage: PermissionsRequestMessage, dataStream?: Readable): Promise<GenericMessageReply>;
+  public async processMessage(tenant: string, rawMessage: PermissionsGrantMessage, dataStream?: Readable): Promise<GenericMessageReply>;
+  public async processMessage(tenant: string, rawMessage: PermissionsRevokeMessage, dataStream?: Readable): Promise<GenericMessageReply>;
+  public async processMessage(tenant: string, rawMessage: RecordsDeleteMessage, dataStream?: Readable): Promise<GenericMessageReply>;
+  public async processMessage(tenant: string, rawMessage: RecordsQueryMessage, dataStream?: Readable): Promise<RecordsQueryReply>;
+  public async processMessage(tenant: string, rawMessage: RecordsReadMessage, dataStream?: Readable): Promise<RecordsReadReply>;
```
Comment (thehenrytsai) [inline src/dwn.ts:103]: Let's remove stream from the parameter for methods that don't require them, such as:

```suggestion
  public async processMessage(tenant: string, rawMessage: RecordsDeleteMessage): Promise<GenericMessageReply>;
  public async processMessage(tenant: string, rawMessage: RecordsQueryMessage: Promise<RecordsQueryReply>;
  public async processMessage(tenant: string, rawMessage: RecordsReadMessage): Promise<RecordsReadReply>;
```

File: src/dwn.ts [hunk 2]
```diff
@@ -90,6 +91,19 @@ export class Dwn {
...
+  public async processMessage(tenant: string, rawMessage: RecordsWriteMessage, dataStream?: Readable): Promise<GenericMessageReply>;
+  public async processMessage(tenant: string, rawMessage: GenericMessage, dataStream?: Readable): Promise<GenericMessageReply>;
+  public async processMessage(tenant: string, rawMessage: unknown, dataStream?: Readable): Promise<UnionMessageReply>;
   public async processMessage(tenant: string, rawMessage: any, dataStream?: Readable): Promise<UnionMessageReply> {
```
Comment (thehenrytsai) [inline src/dwn.ts:107]: Super nit. what about we do the below and remove line 105?

```suggestion
  public async processMessage(tenant: string, rawMessage: GenericMessage, dataStream?: Readable): Promise<UnionMessageReply> {
```

**Review by thehenrytsai — DISMISSED** [review]
💯 🎖️ 🐉 🤯 

Give a medal to this girl right now! This is sick and not complicated! Looks like this will most likely make all the "dedicated handling methods" obsolete.

File: src/dwn.ts (hunk 1 above)
Comment (diehuxx) [reply src/dwn.ts]: Good call! Updated

File: src/dwn.ts (hunk 2 above)
Comment (diehuxx) [reply src/dwn.ts]: Updated

**Review by thehenrytsai — APPROVED** [review]
💯

**Pull request merged**